repos/os/src/drivers/platform/spec/x86/pci_config_access.h

```diff
@@ -141,29 +138,29 @@ namespace Platform {
 			           unsigned value, Device::Access_size size,
 			           bool track = true)
 			{
-				/* write target address */
-				_io_port<REG_ADDR>()->outl(REG_ADDR, _cfg_addr(bus, device,
-				                                               function, addr));
+				const addr_t mmconf = _base + _dev_base(bus, device, function);
+				Attached_io_mem_dataspace mmio(_env, mmconf, MMCONF_REGION_SIZE);
```

Comment:
Please open this Attached_io_mem dataspace once in the constructor and don't create and destruct this io_mem session with every read/write.

Reply:
This is a per-device dataspace. Can you please elaborate?

Reply:
Mmh, unfortunately this class is not per device instead for all pci devices. Before your proposal change this made sense, since there was only one specific io port range for all pci devices. Now it changed.

This "_base + _dev_base()" construct assumes, that there is only one contiguous memory range for a given mmconf range - so it would make sense to open this dataspace once and attach all once.